### DISCUSSION TITLE
fail to configure by CMAKE on 3.3

### DISCUSSION THREAD

**Discussion by shijunz** [audacity/audacity#4599] [Q&A]
### Bug description

Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
The C compiler identification is MSVC 19.35.32216.1
The CXX compiler identification is MSVC 19.35.32216.1
Detecting C compiler ABI info
Detecting C compiler ABI info - done
Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.35.32215/bin/Hostx64/x64/cl.exe - skipped
Detecting C compile features
Detecting C compile features - done
Detecting CXX compiler ABI info
Detecting CXX compiler ABI info - done
Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.35.32215/bin/Hostx64/x64/cl.exe - skipped
Detecting CXX compile features
Detecting CXX compile features - done
Found Python3: C:/Users/hostname/AppData/Local/Programs/Python/Python311/python.exe (found version "3.11.3") found components: Interpreter 
The ASM_NASM compiler identification is YASM
Found assembler: C:/msys64/usr/bin/yasm.exe
Build Info:
  Host System: Windows-10.0.19045
  Host System Name: Windows
  Host System Processor: AMD64
  Host System Version: 10.0.19045
  Host System Architecture: 64-bit

  Compiler: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.35.32215/bin/Hostx64/x64/cl.exe
  Compiler Version: 19.35.32216.1
  Compiler Standard: 17
  Compiler Standard Required: ON
  Compiler Extensions: 

  MSVC Version: 1935
  MSVC Toolset: 143

  Current Commit: caeb58e9a

Check size of int
Check size of int - done
Check size of long
Check size of long - done
Check size of float
Check size of float - done
========== Using Conan version of ZLIB ==========
========== Using Conan version of EXPAT ==========
========== Using Conan version of PNG ==========
========== Using Conan version of JPEG ==========
========== Using Conan version of wxWidgets ==========
========== Using Conan version of libmp3lame ==========
========== Using Conan version of mpg123 ==========
========== Using Conan version of libid3tag ==========
========== Using Conan version of WavPack ==========
========== Using Conan version of Ogg ==========
========== Using Conan version of FLAC ==========
========== Using Conan version of Opus ==========
========== Using Conan version of Vorbis ==========
========== Using Conan version of SndFile ==========
========== Using Conan version of PortAudio ==========
========== Using Conan version of PortMidi ==========
========== Using Conan version of vst3sdk ==========
========== Using Conan version of Catch2 ==========
Using Conan version 1.59.0 (conan.exe)

Removing old audacity remote...
Removing old conan-center-cache remote...
Adding audacity-binaries remote...
Updating global Conan config...

Configuration (profile_host):

[settings]
arch=x86_64
build_type=Debug
compiler=Visual Studio
compiler.runtime=MDd
compiler.version=17
cppstd=17
os=Windows
&:build_type=Debug
[options]
&:use_catch2=True
&:use_expat=True
&:use_flac=True
&:use_libid3tag=True
&:use_libjpeg-turbo=True
&:use_libmp3lame=True
&:use_libpng=True
&:use_libsndfile=True
&:use_mpg123=True
&:use_ogg=True
&:use_opus=True
&:use_portaudio=True
&:use_portmidi=True
&:use_vorbis=True
&:use_vst3sdk=True
&:use_wavpack=True
&:use_wxwidgets=True
&:use_zlib=True
[build_requires]
[env]

Configuration (profile_build):
[settings]
arch=x86_64
build_type=RelWithDebInfo
compiler=Visual Studio
compiler.runtime=MD
compiler.version=17
cppstd=17
os=Windows
&:build_type=Release
[options]
[build_requires]
[env]

ERROR: Error loading conanfile at 'C:\Users\hostname\source\repos\audacity\audacity-git\conan\conanfile.py': Unable to load conanfile in C:\Users\hostname\source\repos\audacity\audacity-git\conan\conanfile.py
  File "<frozen importlib._bootstrap>", line 684, in _load
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\hostname\source\repos\audacity\audacity-git\conan\conanfile.py", line 1, in <module>
    from dataclasses import dataclass
ModuleNotFoundError: No module named 'dataclasses'

Restoring global Conan config...

Traceback (most recent call last):

  File "C:\Users\hostname\source\repos\audacity\audacity-git\conan\conan_runner.py", line 339, in <module>

    subprocess.check_call(conan_options)

  File "C:\Users\hostname\AppData\Local\Programs\Python\Python311\Lib\subprocess.py", line 413, in check_call

    raise CalledProcessError(retcode, cmd)

subprocess.CalledProcessError: Command '['conan.exe', 'install', 'C:\\Users\\hostname\\source\\repos\\audacity\\audacity-git\\conan', '--build=missing', '--install-folder', 'C:/Users/hostname/source/repos/audacity/build-3.3-intel-x64-a2\\conan-install-debug', '--output-folder', 'C:/Users/hostname/source/repos/audacity/build-3.3-intel-x64-a2', '--remote', 'audacity-binaries', '--profile:build', 'C:/Users/hostname/source/repos/audacity/build-3.3-intel-x64-a2\\profile-build.profile', '--profile:host', 'C:/Users/hostname/source/repos/audacity/build-3.3-intel-x64-a2\\profile-host-debug.profile']' returned non-zero exit status 1.

CMake Error at cmake-proxies/cmake-modules/AudacityDependencies.cmake:202 (message):
  Conan failed to install dependencies (1)
  C:/Users/hostname/AppData/Local/Programs/Python/Python311/python.exe
Call Stack (most recent call first):
  CMakeLists.txt:572 (include)

Configuring incomplete, errors occurred!

### Steps to reproduce

1 , download the 3.3.0
2   mkdir build && cd build
3  cmake in the windows OS
4  found the error ----

  File "C:\Users\hostname\source\repos\audacity\audacity-git\conan\conanfile.py", line 1, in <module>
    from dataclasses import dataclass
ModuleNotFoundError: No module named 'dataclasses'

### Expected behavior

configure ok

### Audacity Version

latest beta

### Operating system

Windows

**Comment by crsib** [comment]
`dataclasses` is a part of Python stdlib. It seems that something is wrong with your Python installation

**Reply by shijunz** [reply]
It is not reasonable, after pip install dataclass again, I write test code  for, 
from dataclasses import dataclass

@dataclass
class Person:
    name: str
    age: int
    email: str = ""

person = Person("Alice", 25, "alice@example.com")
print(person)

and I get the right output, so I am sure my python is ok. but the error again:




ERROR: Error loading conanfile at 'C:\Users\shijunz\source\repos\audacity\audacity-git\conan\conanfile.py': Unable to load conanfile in C:\Users\shijunz\source\repos\audacity\audacity-git\conan\conanfile.py
  File "<frozen importlib._bootstrap>", line 684, in _load
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\shijunz\source\repos\audacity\audacity-git\conan\conanfile.py", line 1, in <module>
    from dataclasses import dataclass
ModuleNotFoundError: No module named 'dataclasses'


Restoring global Conan config...

Traceback (most recent call last):

[conanfilepy_dataclasses_module.txt](https://github.com/audacity/audacity/files/11322258/conanfilepy_dataclasses_module.txt)

**Reply by shijunz** [reply]
> `dataclasses` is a part of Python stdlib. It seems that something is wrong with your Python installation

anyway, thank you very much for your quick reply.

**Reply by Q1IQ** [reply]
I ran into the exact same issue, and rolling back the Conan version can solve it for me. Run `pip3 install conan==1.59.0`.
